code/renderers/vue3/src/decorateStory.ts

```diff
@@ -46,25 +47,31 @@ export function decorateStory(
 ): LegacyStoryFn<VueRenderer> {
   return decorators.reduce(
     (decorated: LegacyStoryFn<VueRenderer>, decorator) => (context: StoryContext<VueRenderer>) => {
-      let story: VueRenderer['storyResult'] | undefined;
-
+      let story: VueRenderer['storyResult'] = { isNull: true };
       const decoratedStory: VueRenderer['storyResult'] = decorator((update) => {
         story = decorated({
           ...context,
           ...sanitizeStoryContextUpdate(update),
         });
+        const argsChanged =
+          update && update.args && Object.keys(update).length === 1 && !isVNode(story);
+        // TODO: this is a hack to avoid re-rendering the story when the args are not changed
+        // we should find a better way to do this
+        // i should get only update = { args: { ...context.args, text:... } } from the decorator and not the whole context
+        if (argsChanged) {
+          story = h(story, update.args);
+        }
         return story;
       }, context);
 
-      if (!story) {
-        story = decorated(context);
-      }
+      if (story.isNull) story = decorated(context);
 
       if (decoratedStory === story) {
         return story;
       }
-
-      return prepare(decoratedStory, h(story, context.args)) as VueRenderer['storyResult'];
+      const props = story.props ?? context.args;
+      const innerStory = () => h(story, props);
```

Comment:
here the inner Story should be a function to be binded with the props changes, this makes CSF2  format works too